components/org.wso2.carbon.identity.api.server.organization.user.invitation.management/org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/user/invitation/management/v1/core/GuestApiServiceCore.java

```diff
@@ -248,16 +252,25 @@ private String includeData(UserInvitationMgtConstants.ErrorMessage error, String
         return error.getDescription();
     }
 
-    private InvitationSuccessResponse createInvitationSuccessResponse(Invitation invitation) {
+    private List<InvitationSuccessResponse> createInvitationSuccessResponse(List<Invitation> invitationList) {
 
-        InvitationSuccessResponse invitationSuccessResponse = new InvitationSuccessResponse();
-        invitationSuccessResponse.setUsername(invitation.getUsername());
-        invitationSuccessResponse.setEmail(invitation.getEmail());
-        if (invitation.getRoleAssignments().length > 0) {
-            List<RoleAssignmentResponse> roleAssignmentResponses = buildRoleAssignmentResponse(invitation);
-            invitationSuccessResponse.setRoles(roleAssignmentResponses);
+        List<InvitationSuccessResponse> invitationSuccessResponseList = new ArrayList<>();
+        for (Invitation invitation : invitationList) {
+            if (invitation.getSkippedUsersList().isEmpty()) {
```

Comment:
Do we have the skipped user list in every invitation object?

Reply:
Let's assume we have the skipped user list in every invitation object, then we cannot gurantee that it will not come in the first object of the invitation list. So here we are doing some unwanted looping here.

Reply:
Here we expect to create multiple invitations at once. Hence there can be a chance either all of the invitations will get created successfully or one/few of them failed due to any invalidation in the request.
- If it is a successful creations then the response will contain the created invitation details for each user - Refer https://github.com/wso2/product-is/issues/16719#issuecomment-1833886781
- If there any failure happened, invitation creation for that particular user will be skipped and the user will be added to a list and finally there will be list return with an invitation object with the skipped user's list - Refer https://github.com/wso2-extensions/identity-organization-management/pull/315/files#diff-ebba2067b3fa55047938aa6006a6864220dd3d4de3854fc2894d203714e0e926R155-R170
- Hence, in this list there will be one invitation object and if it has skipped user's list, just return the response with the skipped user's list. There won't be more iterations in this case.